cli/smartcontract/smart_contract.go

```diff
@@ -819,6 +836,77 @@ func contractDeploy(ctx *cli.Context) error {
 	return nil
 }
 
+// contractUpdate updates an existing smart contract.
+func contractUpdate(ctx *cli.Context) error {
+	var (
+		paramsStart   = 0
+		args          []string
+		nefBytes      []byte
+		manifestBytes []byte
+		nefParam      any
+		manifestParam any
+	)
+	args = ctx.Args().Slice()
+	if len(args) < 1 {
+		return cli.Exit(errNoScriptHash, 1)
+	}
+	contractHash, err := flags.ParseAddress(args[0])
+	if err != nil {
+		return cli.Exit(fmt.Errorf("invalid contract hash '%s': %w", args[0], err), 1)
+	}
+	paramsStart++
+
+	if ctx.String("in") != "" {
+		_, nefBytes, err = readNEFFile(ctx.String("in"))
+		if err != nil {
+			return cli.Exit(fmt.Errorf("failed to read .nef file: %w", err), 1)
+		}
+		nefParam = nefBytes
+	}
+	if ctx.String("manifest") != "" {
+		_, manifestBytes, err = readManifest(ctx.String("manifest"), contractHash)
+		if err != nil {
+			return cli.Exit(fmt.Errorf("failed to read manifest file: %w", err), 1)
+		}
+		manifestParam = manifestBytes
+	}
+	if nefBytes == nil && manifestBytes == nil {
+		return cli.Exit(fmt.Errorf("either manifest or .nef required"), 1)
+	}
+
+	params := []any{nefParam, manifestParam}
+
+	signOffset, data, err := cmdargs.ParseParams(args[paramsStart:], true)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("unable to parse 'data' parameter: %w", err), 1)
+	}
+	if len(data) > 1 {
+		return cli.Exit("'data' should be represented as a single parameter", 1)
+	}
+	if len(data) != 0 {
+		params = append(params, data[0])
+	}
```

Comment:
Let's improve this functionality a bit: if there's no data provided, the first call should be done with `nil` data appended to `params`. If this call fails, then the second attempt should be done with two parameters only.

The user should be able to disable second attempt via `--strict` flag.